### PR TITLE
Update pin-project-lite and fix warnings

### PR DESCRIPTION
**Pull request by sfackler**

### CODE REVIEW

File: zipkin-types/src/span_id.rs [hunk 1]
```diff
@@ -118,7 +118,7 @@ pub struct SpanIdParseError(Option<DecodeError>);
 
 impl fmt::Display for SpanIdParseError {
     fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(fmt, "{}: ", self.description())?;
+        fmt.write_str("error parsing span: ")?;
```
Comment (dtobin) [inline zipkin-types/src/span_id.rs:121]: Why the change here?

Comment (sfackler) [reply zipkin-types/src/span_id.rs:121]: The `Error::description` method is deprecated, and since https://github.com/palantir/rust-zipkin/commit/abbd991b8b18eb65f2dbed163121b4df6c8a9f65 we don't even implement it so this message didn't make sense.

**Review by dtobin — APPROVED** [review]
